src/electron/views/automated-checks/components/title-bar.tsx

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { BrowserWindow } from 'electron';
+import { ActionButton } from 'office-ui-fabric-react/lib/Button';
+import * as React from 'react';
+
+import { NamedFC } from 'common/react/named-fc';
+import { BrandWhite } from 'icons/brand/white/brand-white';
+import { titleBar } from './title-bar.scss';
+
+export type TitleBarDeps = {
+    currentWindow: BrowserWindow;
+};
+
+export interface TitleBarProps {
+    deps: TitleBarDeps;
+}
+
+export const TitleBar = NamedFC<TitleBarProps>('TitleBar', (props: TitleBarProps) => {
+    const currrentWindow = props.deps.currentWindow;
+    const minimize = () => currrentWindow.minimize();
+    const maximize = () => (currrentWindow.isMaximized() ? currrentWindow.restore() : currrentWindow.maximize());
```

Comment:
how about using store here instead? windowstateStore? & let WindowFrameUpdater deal with window updates?

Reply:
That's a very good point... not sure if the the current `WindowFrameUpdated` implementation handles restore/maximize logic as we have here...

Reply:
we can improve the windowFrameUpdater, so that window state is maintained in one place.

Reply:
Sounds like `WindowFrameUpdater` is the way to go. Do we want to get that into this PR, or bring it in as a followup PR to keep PRs more concise?

Reply:
I think it should be part of this PR. @devaradhanm ?

Reply:
Updated to use `WindowFrameUpdater` + `WindowStateStore` for changes to window state